tsl/src/continuous_aggs/union_view.c

```diff
@@ -0,0 +1,299 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+
+#include <postgres.h>
+#include <catalog/pg_type.h>
+#include <nodes/makefuncs.h>
+#include <nodes/nodeFuncs.h>
+#include <nodes/parsenodes.h>
+#include <nodes/primnodes.h>
+#include <optimizer/clauses.h>
+#include <parser/parse_func.h>
+#include <utils/builtins.h>
+#include <utils/typcache.h>
+
+#include "continuous_aggs/create.h"
+#include "extension_constants.h"
+
+#define BOUNDARY_FUNCTION "cagg_watermark"
+#define INTERNAL_TO_DATE_FUNCTION "to_date"
+#define INTERNAL_TO_TSTZ_FUNCTION "to_timestamp"
+#define INTERNAL_TO_TS_FUNCTION "to_timestamp_without_timezone"
```

Comment:
I dont think a cache is warranted here since this only runs when the view itself is modified.

Reply:
The main point of my suggestion is actually not the cache (but it is a nice bonus), but rather a way to structure the code with the function definitions in one place instead of having them spread loosely across the code.

Currently, there are a lot of `Oid argtyp[] = { OIDOID };` type variables spread across the code with no clear connection to the function those argtyps belong to. My suggestion puts these function signature items firmly in one place and strictly associated with each function.

This will lead to much cleaner code even if you don't care about the fact that you are also "remembering" the OIDs you look up.

Reply:
I prefer not to do this for this PR and I think if we want a function cache for this it should be in a more central place to facilitate reuse instead of multiple files having their own.